app/src/main/java/org/apache/taverna/mobile/ui/tutorial/TutorialActivity.java

```diff
@@ -0,0 +1,204 @@
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+
+public class TutorialActivity extends AppCompatActivity {
+
+    @BindView(R.id.view_pager)
+    ViewPager viewPager;
+
+    @BindView(R.id.layoutDots)
+    LinearLayout dotsLayout;
+
+    @BindView(R.id.btn_skip)
+    Button bSkip;
+
+    @BindView(R.id.btn_next)
+    Button bNext;
+
+
+    ViewPagerAdapter viewPagerAdapter;
+    TextView[] dots;
+    int[] layouts;
+    PreferencesHelper preferencesHelper;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        preferencesHelper = new PreferencesHelper(this);
+        if (!preferencesHelper.isFirstTimeLaunch()) {
+            launchFlashScreen();
+            finish();
+        }
+
+        setContentView(R.layout.activity_tutorial);
+
+        ButterKnife.bind(this);
+
+
+        if (Build.VERSION.SDK_INT >= 21) {
+            getWindow().getDecorView()
+                    .setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+        }
+
+        layouts = new int[]{
+                R.layout.tutorial_slide1,
+                R.layout.tutorial_slide2,
+                R.layout.tutorial_slide3,
+                R.layout.tutorial_slide4};
+
+        addBottomDots(0);
+
+        viewPagerAdapter = new ViewPagerAdapter();
+        viewPager.setAdapter(viewPagerAdapter);
+        viewPager.addOnPageChangeListener(viewPagerPageChangeListener);
+
+    }
+
+    @OnClick(R.id.btn_skip)
+    public void skipClick(View v) {
+        launchFlashScreen();
+    }
+
+    @OnClick(R.id.btn_next)
+    public void nextClick(View v) {
+        int current = getItem(+1);
+        if (current < layouts.length) {
+            viewPager.setCurrentItem(current);
+        } else {
+            launchFlashScreen();
+        }
+    }
+
+    public void addBottomDots(int currentPage) {
+        dots = new TextView[layouts.length];
+
+        int[] colorsActive = getResources().getIntArray(R.array.array_dot_active);
+        int[] colorsInactive = getResources().getIntArray(R.array.array_dot_inactive);
+
+        dotsLayout.removeAllViews();
+        for (int i = 0; i < dots.length; i++) {
+            dots[i] = new TextView(this);
+            dots[i].setText(Html.fromHtml("&#8226;"));
+            dots[i].setTextSize(35);
+            dots[i].setTextColor(colorsInactive[currentPage]);
+            dotsLayout.addView(dots[i]);
+        }
+
+        if (dots.length > 0)
+            dots[currentPage].setTextColor(colorsActive[currentPage]);
+    }
+
+    public int getItem(int i) {
+        return viewPager.getCurrentItem() + i;
+    }
+
+    public void launchFlashScreen() {
+        preferencesHelper.setFirstTimeLaunch(false);
+        startActivity(new Intent(TutorialActivity.this, FlashScreenActivity.class));
+        finish();
+    }
+
+    ViewPager.OnPageChangeListener viewPagerPageChangeListener =
```

Comment:
Implement this interface in place of the object.

Reply:
I tried this but we need to have the selected page item in `viewpager.addOnPageChangeListener`